### PR TITLE
Add changlog labels

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
This PR purpose is to raise a discussion. 

The idea is to add two labels related to the changelog.  This should split the job for changelog creation into small steps and make a decision about that when the feature is in progress. 

In short, each PR will be in one of 3 states:  we want to add changes to this PR to the changelog, we don't want it, and not sure (the label is not set).   This kind of decision is much easier to do during the development phase than at the end.  Also, it could be done by the person who is involved in the review/merge process.

On top of that, we could build some simple automation.  If the label is `category:feature` set it automatically to yes.  If this is some technical staff, or refactoring, tweak, make it no. 

Another automation could be to check if the change log file was updated. If we want to have a changelog entry, it's good to write it now.

@Vezzra what do you think?

### CODE REVIEW

**Comment by geoffthemedio** [comment]
> we want to revisit it when making a change log, we don't want it, and not sure (the label is not set)

Not sure what it means to "revisit" a pull request...

**Comment by Cjkjvfnby** [comment]
> Not sure what it means to "revisit" a pull request...

reworded that original text. 

Open PR page and put some information to changelog

**Comment by Vezzra** [comment]
IMO that would be a helpful addition, it's certainly something ***I*** would appreciate and use if I were the one compiling the changelog.

However, as that's a task usually taken care of by @geoffthemedio, and I have to say, very efficiently so, his opinion probably matters most. AFAIK @geoffthemedio's workflow is working through the commit log when compiling the changelog, so the question is, how much help those changelog labels would be for him - @geoffthemedio?

**Comment by geoffthemedio** [comment]
I think I'd just read the commit logs and not care about a tag in github , which I'm guessing won't be enforced in any useful way so won't be really reliable. Not sure if they'd even be visible in the exported logs that I go through, either. I don't really understand how this would work or how it's supposed to help.

The difficult parts of doing the commit logs are usually figuring out if a change is actually relevant in the latest version, or if it was a duplicate or part of another already-noted change or if it was overridden by a later change so doesn't apply in the final version being described, or if it is just a bugfix or tweaks to a change since the previous release so either doesn't need mentioning (since I don't note a bugfix to an issue added since the previous release) or needs mentioning but just by updating the values in a previous note (ie. damage to a weapon was changed 3 times over 8 months... I just want to note the final value, not all three separately).

**Comment by Cjkjvfnby** [comment]
> I think I'd just read the commit logs and not care about a tag in github , which I'm guessing won't be enforced in any useful way so won't be really reliable.

The idea is to review PR by PR.

Benefits:
- A smaller number of PR than commits
- PR usually has more context on the page, than commit in the git log
- You could filter a part of PRs using search filters https://github.com/freeorion/freeorion/pulls?q=is%3Apr+is%3Amerged+not%3A+-label%3Acategory%3Atweak+-label%3Acategory%3Arefactoring  (and we could simplify it by adding `changlelog:yes` label with the GitHub actions)
- You could sort the changelog in an iterative manner, find all PR that does not have the label, and set it. 
- We could use GitHub action that will fail if `changlelog:yes` and the changelog file in the repo is not updated.
- Instead of doing the job after many months after the changes you could do it before merge, when you and the author have much better context.

This won't solve the problem of the things that were changed multiple times during release, but also won't make it more complex.  
Filtering PRs by component might help with the narrowing scope where you looking for PRs that might require changelogs merged.  Also, you might notice that the change affects the previous one when you doing it, and adjust the changelog and label.

In short, this will make working with changelog split across the timeline and a part of your work could be delegated to the PR authors.
